localstack/services/s3/provider.py

```diff
@@ -255,6 +255,16 @@ def get_bucket_location(
         self, context: RequestContext, bucket: BucketName, expected_bucket_owner: AccountId = None
     ) -> GetBucketLocationOutput:
         response = call_moto(context)
+        # LocationConstraint is a bit of weird response, we return directly XML from moto because it does not have
+        # a GetBucketLocationOutput XML tag.
+        # some tools (Serverless) require a newline after the "<?xml ...>\n" preamble line, e.g., for LocationConstraint
+        # this is required because upstream moto is generally collapsing all S3 XML responses
+        location_constraint_xml = response["LocationConstraint"]
```

Comment:
This seems to be a bit fragile in case of an exception. Does this work for the error cases as well? f.e. if the bucket does not exist?

Reply:
It will, because the exception will be raises at `call_moto`, and this code won't be called if there's no positive response from `call_moto`. It can however be fragile in case of a moto response change, but we should see it coming with the integration test while bumping the version